src/screens/ChangeWalletModal.js

```diff
@@ -0,0 +1,201 @@
+import React, { useCallback } from 'react';
+import { View } from 'react-native';
+import { useNavigation } from 'react-navigation-hooks';
+import { useDispatch } from 'react-redux';
+import WalletList from '../components/change-wallet/WalletList';
+import { Modal } from '../components/modal';
+import WalletTypes from '../helpers/walletTypes';
+import { useAccountSettings, useInitializeWallet, useWallets } from '../hooks';
+import {
+  addressSetSelected,
+  createAccountForWallet,
+  walletsSetSelected,
+  walletsUpdate,
+} from '../redux/wallets';
+import { logger } from '../utils';
+
+const walletRowHeight = 54;
+
+const ChangeWalletModal = () => {
+  const { wallets, selected: selectedWallet } = useWallets();
+
+  const { goBack, navigate } = useNavigation();
+  const dispatch = useDispatch();
+  const { accountAddress } = useAccountSettings();
+  const initializeWallet = useInitializeWallet();
+  let rowsCount = 0;
+  if (wallets) {
+    Object.keys(wallets).forEach(key => {
+      // Wallet header
+      // If there's more than account we group them by wallet
+      if (wallets[key].addresses.length > 1) {
+        rowsCount += 1;
+      }
+      // Addresses
+      rowsCount += wallets[key].addresses.filter(account => account.visible)
+        .length;
+
+      // Add account
+      if (
+        [WalletTypes.mnemonic, WalletTypes.seed].indexOf(wallets[key].type) !==
+        -1
+      ) {
+        rowsCount += 1;
+      }
+    });
+    // Import wallet
+    rowsCount += 1;
+  }
+  let listHeight = walletRowHeight * rowsCount;
+  if (listHeight > 298) {
+    listHeight = 298;
+  }
+
+  const onChangeAccount = useCallback(
+    async (wallet_id, address) => {
+      try {
+        const wallet = wallets[wallet_id];
+        dispatch(walletsSetSelected(wallet));
+        dispatch(addressSetSelected(address));
+        await initializeWallet();
```

Comment:
Currently, switching back and forth between accounts will create more Zerion connections (with the old ones still listening to their messages too.) If you go from wallet A to wallet B back to wallet A, you'll end up with 3 websockets, 2 for A, 1 for B. We can discuss how we want to manage these ideally. We could share the same websocket and have it subscribe to multiple addresses, we'll just need to change how we handle subscribe/unsubscribe and the different redux states for the different addresses.